cabby/rvs/generate_rvs.py

```diff
@@ -73,63 +73,66 @@ def main(argv):
   for entity_idx, entity in enumerate(entities):
     current_templates = gen_templates.copy()  # Candidate templates.
 
+
     # Use only landmarks that have a main tag.
     for landmark_type, landmark in entity.geo_landmarks.items():
-      if landmark_type not in current_templates:
+      if landmark_type not in current_templates or landmark_type == "start_point":
         continue
-      if landmark.main_tag is '':
+      if not landmark.main_tag:
         current_templates = current_templates[
           current_templates[landmark_type] == False]
       else:
         current_templates = current_templates[current_templates[landmark_type] == True]
 
     # Use features that exist.
     for feature_type, feature in entity.geo_features.items():
-      if feature_type not in current_templates:
+
+      if feature_type not in current_templates or \
+        feature_type=='intersections':
         continue
-      if int(feature) == -1:
-        # print(current_templates[current_templates['intersections'] == False][['intersections', 'blocks']])
-        # print (feature_type)
+      if not feature:
         current_templates = current_templates[
           current_templates[feature_type] == False]
       else:
         current_templates = current_templates[current_templates[feature_type] == True]
 
-      if int(entity.geo_features['intersections']) > 0:
+    intersection = entity.geo_features['intersections']
+    intersection = -1 if intersection is None else int(intersection)
 
-        # Pick templates with either blocks or intersections.
-        is_block = randint(0, 1)
-        blocks = int(entity.geo_features['intersections']) - 1 \
-          if is_block else -1
-        if int(entity.geo_features['intersections']) == 1:
-          # Filter out templates without the next intersection mention.
-          current_templates = current_templates[
-            current_templates['next_intersection'] == True]
-        elif blocks == 1:
-          # Filter out templates without the next block mention.
-          current_templates = current_templates[
-            current_templates['next_block'] == True]
-        else:
-          if blocks > 1:
-            # Filter out templates without mentions of the number of blocks
-            # that should be passed.
-            current_templates = current_templates[
-              current_templates['blocks'] == True]
-          else:
-            # Filter out templates without mentions of the number of
-            # intersections that should be passed.
-            current_templates = current_templates[
-              current_templates['intersections'] == True]
-      else:
+    if intersection > 0:
 
-        # Filter out templates with mentions of intersection\block.
+      # Pick templates with either blocks or intersections.
+      is_block = random.randint(0, 1)
+      blocks = intersection - 1 if is_block else -1
```

Comment:
I would put this after the first check for whether intersection == 1, so it would be more like:

```
if intersection == 1:
  <stuff>
else:
  if random.randint(0,1):
    if intersection == 2: # one block
      <stuff>
   else: # multiple blocks
      <stuff>
  else:
    <multiple intersections template>
```

Ie, no need for blocks variables, but probably good to add some 2-3 word docstrings to clarify.